.github/workflows/publish_charm.yaml

```diff
@@ -33,36 +34,25 @@ on:
         description: Charmcraft channel to use for publishing the charm and resources
         type: string
         default: latest/stable
-
+      workflow-run-id:
+        description: >-
+          Use the newly built charms and images in this workflow run as the new version to upload. 
+          If these are not provided, the system will default to using charms and rocks from the most 
+          recent successful integration test that matches the git tree ID.
+        type: string
+        default: ""
 env:
   REGISTRY: ghcr.io
   OWNER: ${{ github.repository_owner }}
 
 jobs:
-  branch-up-to-date-check-enabled:
```

Comment:
I don't see this as part of the action

Reply:
The intent of this step was to prevent pushing artifacts that might not contain all the changes. I'll illustrate with an example, it's easier:
- We have to open PRs with successful CI checks
- PR A is merged and the artifacts are retrieved from the build and pushed to charmhub. -> OK
- PR B is merged and the artifacts are retrieved from the build and pushed to charmhub. If the check is disabled, the code does not have to be rebased, hence the arctifacts do not contain feature A. If the check is enabled, the PR will have to be rebased, triggering a new build from which the artifacts will be fetched

Reply:
Okay, I understand what you're saying. This issue is resolved because the publish workflow only retrieves artifacts that share the same git tree id. Since the file contents is always the same when two commits have the same git tree id, this ensures that the artifact is constructed from identical source code. In the case of non-linear history repositories, the publish workflow will just fail.

Reply:
Understood

Reply:
I update the error message to recommend enable branch update check if it can't find the matching tree id, hope this can help

https://github.com/canonical/operator-workflows/blob/679bfe7624df6957adc37bc3b7b326fc02f3032c/src/publish.ts#L79-L82